### DISCUSSION TITLE
Problems with FlxG.cameras.setDefaultDrawTarget

### DISCUSSION THREAD

**Discussion by Lbones-ai** [HaxeFlixel/flixel#2347] [Q&A] [answered]
Hi everyone,
since the new version of flixel I have a problem with a deprecated command: FlxCamera.defaultCameras, in the api is suggested to use another command:

> Deprecated: "`FlxCamera.defaultCameras` is deprecated, use `FlxG.cameras.setDefaultDrawTarget` instead"

Here is my code:

```haxe
_gameCamera = new FlxCamera(0, 0, FlxG.width, FlxG.height);
_cameraUI = new FlxCamera(0, 0, FlxG.width, FlxG.height);

_gameCamera.bgColor = FlxColor.BLACK;
_cameraUI.bgColor = FlxColor.TRANSPARENT;

CAMERA_SCALE = 1.6;
_gameCamera.zoom = CAMERA_SCALE;

FlxG.cameras.reset(_gameCamera);
FlxG.cameras.add(_cameraUI);

FlxCamera.defaultCameras = [_gameCamera];                     //zoom works
//FlxG.cameras.setDefaultDrawTarget(_gameCamera, true);  //zoom doesn't work
```

The instruction:
`FlxCamera.defaultCameras = [_gameCamera];`
keeps the _gameCamera zoom at 1.6, but it's deprecated

The instruction:
`FlxG.cameras.setDefaultDrawTarget(_gameCamera, true);`
doesn't keep the zoom of _gameCamera, and any subsequent _gameCamera.zoom command is ignored

The only way for the zoom to work is using the old deprecated command, any idea why is that?

**Comment by Geokureli** [comment, marked as answer]
@Lbones-ai, the problem may be that you're adding _cameraUI as a defaultDraw camera, too, so it's drawing everything at zoom:1.0 after/on top of the _gameCamera, try the following and setDefaultDrawTarget should work as expected.
```hx
FlxG.cameras.add(_cameraUI, false);
```
Also, you shouldn't need to even call `setDefaultDrawTarget(_gameCamera, true);` as `reset(_gameCamera)` will set it as a default draw target

**Reply by Lbones-ai** [reply]
That was it, I forgot that the default was True 😓

Thank you!

**Reply by Dramasso** [reply]
where do i put this?

**Reply by Geokureli** [reply]
can you be more specific about what you're trying to do

**Reply by Dramasso** [reply]
so is the code you put in required in some part of kade engine or not and if so where do i put it

**Reply by Dramasso** [reply]
like do i need to put FlxG.cameras.add(_cameraUI, false); somewhere

**Reply by Geokureli** [reply]
In LBones case he wanted 2 cameras, 1 for the game and 1 for the UI, so that he could zoom in the game without zooming in the UI. is there something specific you're trying to with cameras?

**Reply by Dramasso** [reply]
well before the game starts i get the FlxG.cameras.setDefaultDrawTarget error then it crashes at 20/21 loading

**Reply by Geokureli** [reply]
I would just create a new discussion thread explaining what you're trying to do, what code you used and what error it's giving you. I'll try to help you there
